qiskit/pulse/transforms.py

```diff
@@ -375,6 +379,85 @@ def align_sequential(schedule: Schedule) -> Schedule:
     return aligned
 
 
+def align_equispaced(schedule: Schedule,
+                     duration: int) -> Schedule:
+    """Schedule a list of pulse instructions with equivalent interval.
+
+    Args:
+        schedule: Input schedule of which top-level ``child`` nodes will be
+            reschedulued.
+        duration: Duration of context. This should be larger than the schedule duration.
+
+    Returns:
+        New schedule with input `schedule`` child schedules and instructions
+        aligned with equivalent interval.
+
+    Notes:
+        This context is convenient for writing PDD or Hahn echo sequence for example.
+    """
+    if duration and duration < schedule.duration:
+        return schedule
+    else:
+        total_delay = duration - schedule.duration
+
+    if len(schedule._children) > 1:
+        # Calculate the interval in between sub-schedules.
+        # If the duration cannot be divided by the number of sub-schedules,
+        # the modulo is appended and prepended to the input schedule.
+        interval, mod = np.divmod(total_delay, len(schedule._children) - 1)
+    else:
+        interval = 0
+        mod = total_delay
+
+    # Calculate pre schedule delay
+    delay, mod = np.divmod(mod, 2)
+
+    aligned = Schedule()
+    # Insert sub-schedules with interval
+    _t0 = int(aligned.stop_time + delay + mod)
```

Comment:
What happens if the modulo is not evenly divisible? I don't see how it is appended evenly to the end of the schedule.

Reply:
I think we cannot divide odd duration value evenly so the 1 tick of modulo comes in the beginning (no more than 1). Perhaps we can add optional argument that specifies the preference on either `'beginning'` or `'end'`.

Reply:
The behavior you describe is fine, I just wasn't confident it was doing this.